build/oil-defs/Python-2.7.13/Modules/posixmodule.c/posix_methods.def

```diff
@@ -23,6 +23,7 @@ static PyMethodDef posix_methods[] = {
   {"getpid", posix_getpid, METH_NOARGS},
   {"getuid", posix_getuid, METH_NOARGS},
   {"getppid", posix_getppid, METH_NOARGS},
+  {"getpgid", posix_getpgid, METH_VARARGS},
```

Comment:
probably a copy/paste bug from `getppid` above? `posix_getpgid()` in `pyext/posixmodule.c` takes a a single int arg

Reply:
Hm I'm not sure what you mean? 

These files should actually be auto-generated, so you can leave them out.  There is a weird procedure documented at the top of  `build/cpython-defs.sh`.

What it does is actually lex all the source code, and then if it sees the token `getpgid`, that function will automatically be copied be copied from `Python-2.7.13/` to this file.

So when you actually add those calls, it should "work automatically".


(This was a thing I did a long time ago to "strip down" the build, which ended up being a bit of a dead end.  Long term I want to get rid of the whole "OVM tarball" build.)

Reply:
Oh weird. `Python-2.7.13/Modules/posixmodule.c` (line 9000) uses `METH_VAARGS` for this. I wonder what went wrong.

My local build was failing because of a mismatched type signature without this change, but maybe I missed a step... I did the usual `build/py.sh all` thing